src/main/java/io/pravega/perf/ReaderWorker.java

```diff
@@ -47,81 +58,78 @@ public Void call() throws InterruptedException, ExecutionException, IOException
         return null;
     }
 
-    private class EventsReader implements Performance {
-
-        public void benchmark() throws IOException {
-            String ret = null;
-            try {
-                for (int i = 0; i < events; i++) {
-                    final long startTime = System.currentTimeMillis();
-                    ret = readData();
-                    if (ret != null) {
-                        stats.recordTime(startTime, System.currentTimeMillis(), ret.length());
-                    }
+
+    public void EventsReader() throws IOException {
+        String ret = null;
+        try {
+            int i = 0;
+            while (i < events) {
+                final long startTime = System.currentTimeMillis();
+                ret = readData();
+                if (ret != null) {
+                    stats.recordTime(startTime, System.currentTimeMillis(), ret.length());
+                    i++;
                 }
-            } finally {
-                close();
             }
+        } finally {
+            close();
         }
     }
 
-    private class EventsReaderRW implements Performance {
-        public void benchmark() throws IOException {
-            String ret = null;
-            try {
-                for (int i = 0; i < events; i++) {
-                    ret = readData();
-                    if (ret != null) {
-                        final long endTime = System.currentTimeMillis();
-                        final long startTime = Long.parseLong(ret.substring(0, TIME_HEADER_SIZE));
-                        stats.recordTime(startTime, endTime, ret.length());
-                    }
+
+    public void EventsReaderRW() throws IOException {
+        String ret = null;
+        try {
+            int i = 0;
+            while (i < events) {
```

Comment:
Same here about `for` instead of `while`.

Reply:
increment happens on a condition ; hence i changed it to while